File: src/main/java/de/ng/ahnenwin2gedcom/csv/Verbindung.java
package de.ng.ahnenwin2gedcom.csv;

import java.util.Arrays;

public enum Verbindung {
    ANDERE_BEZIEHUNG("andere Beziehung"),
    EHESCHLIESSUNG("Eheschliessung"),
    LEBENSGEMEINSCHAFT("Lebensgemeinschaft"),
    VERLOBUNG("Verlobung"),
    LEER("");

    private final String csvValue;

    Verbindung(String csvValue) {
        this.csvValue = csvValue;
    }

    static Verbindung findByCsvValue(String csvValue) {
        String trimmedCsvValue = csvValue == null ? "" : csvValue.trim();
        return Arrays.stream(values())
                .filter(verbindung -> trimmedCsvValue.equals(verbindung.csvValue))
                .findAny()
                .orElse(Verbindung.LEER);
    }
}

File: src/main/java/de/ng/ahnenwin2gedcom/hej/Verbindung.kt
package de.ng.ahnenwin2gedcom.hej

enum class Verbindung(private val csvValue: String) {
    ANDERE_BEZIEHUNG("andere Beziehung"),
    EHESCHLIESSUNG("Eheschliessung"),
    LEBENSGEMEINSCHAFT("Lebensgemeinschaft"),
    VERLOBUNG("Verlobung"),
    LEER("");

    internal companion object {
        @JvmStatic
        fun findByCsvValue(csvValue: String?): Verbindung {
            val trimmedCsvValue = csvValue?.trim () ?: ""
            return values().firstOrNull{ it.csvValue == trimmedCsvValue } ?: LEER
        }
    }
}
